### DISCUSSION TITLE
[HELP] Stripe customer id to Supabase doesn't work

### DISCUSSION THREAD

**Discussion by charlesoestergaard** [supabase/supabase#5843] [Questions]
I was following [Jon Meyers](https://egghead.io/instructors/jon-meyers) guide on egghead to implement Stripe to my supabase project.

The customer gets created in stripe, but the customer id doesnt get set in my supabase project.

Mabye Supabase Function hook is not triggering on Insert ?
(edit: Tried to disable email confirmation, but it dident help)
The user will register on my page, and have to confirm email to login? <- Can this be the issue? In the turotial he instantly logs in without register.

API file (Table is called "profiles" column is called "stripe_customer" 
![Code_ucPiQJQ0wz](https://user-images.githubusercontent.com/59709554/157039057-b0a19106-6b16-4914-9019-ec98c10bdab8.png)

.env.local file
![Code_X76PdhRTDK](https://user-images.githubusercontent.com/59709554/157039193-7d337dc2-47ce-418d-8d80-031da4cdfeb8.png)

Supabaes (Profiles) table

![chrome_S6BmgcxvUX](https://user-images.githubusercontent.com/59709554/157039964-a6964814-592c-4abf-b771-9a71233e4233.png)
g…]()

Supabase Function Hook
![Code_O5ALDDuYF0](https://user-images.githubusercontent.com/59709554/157039673-33861e2c-fb75-4c74-bedb-e3a249f5f53e.png)

**Comment by AlexanderCryptoOwl** [comment]
Hello, fellow dane,

Jeg skriver på engelsk det hurtigere...

I don't have this setup so can't help a loot. I assume a row in the profile tables gets inserted once the user signups successfully, so shouldn't be an issue with email confirmation? 

What is the console.log of const customer?

Also, you are doing let {error}, not sure it matters but I always use and see const { data, error }

**Reply by charlesoestergaard** [reply]
I dont get any console.log, which is weird. Const or let doesn't matter for this particular part of the code, also i removed it so its just 

```  await supabase
    .from("profiles")
    .update({ stripe_customer: customer.id })
    .match("id", req.body.record.id);
```
    
I found out that the email confirm dident change anything.

**Comment by GaryAustin1** [comment]
Make sure rec.body.record.id contains a valid uuid (you can also verify in the network tab for the request).  
Are you getting in network errors?
You also appear to have RLS enabled.  For update you need a using policy, or a using and with check policy that will return true.
It is also possible you don't have a signed in user when you make the update call.  You can check that by temporarily setting your RLS to user.role() = 'anon'.  If that works you are not logged in.

**Comment by danielcranney** [comment]
Hi @rumfiske,
Did you ever find the solution to this issue? I've run into the same problem...

**Reply by charlesoestergaard** [reply]
Yes it was becuase i had RLS enabled. But you Can also fix it, if you use the other API key from supabase.

**Reply by danielcranney** [reply]
Thanks a lot! Where did you use the other key?

**Reply by charlesoestergaard** [reply]
![DFDFFC2B-CBE3-422C-90D5-572D23868433](https://user-images.githubusercontent.com/59709554/161503102-8d335a6b-f95c-4524-b087-12f812a2d88f.png)

see this key Can overwrite RLS. Use that key
